### PR TITLE
feat: Add servicex client

### PR DESCRIPTION
**Pull request by matthewfeickert**
* c.f. https://github.com/ssl-hep/ServiceX_frontend
* Recipe generated with grayskull via 'grayskull pypi servicex' and then edited to add additional information.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [N/A] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by matthewfeickert** [comment]
https://github.com/conda-forge/func-adl-feedstock/pull/9 should fix the current errors.

**Comment by matthewfeickert** [comment]
@conda-forge-admin, please restart ci

**Comment by matthewfeickert** [comment]
(Having to force push as the CI lost the branch on restart through the bot)

File: recipes/servicex/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/servicex-{{ version }}.tar.gz
+  sha256: 10fe7aa45ade9ba978563e20cf51f80aa149139423fd0b0766cd7e72e06e1f74
+
+build:
+  skip: true  # [py<38]
```
Comment (matthewfeickert) [inline recipes/servicex/meta.yaml:13]: This is a `noarch` package in reality, but I believe that to use to use things like `# [py<310]` (I forget the technical term that conda-forge uses here) you can't use them with `noarch`.

**Comment by matthewfeickert** [comment]
@conda-forge/help-python, ready for review!

File: recipes/servicex/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipes/servicex/meta.yaml:13]: Just remove the selectors, installing those packages in py>39/py310 is not a problem.

File: recipes/servicex/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    - aiohttp-retry >=2.8.3
+    - httpx >=0.24
+    - miniopy-async >=1.20.1
+    - tinydb >=4.7
+    - google-auth >=2.17
+    - typer >=0.12.1
+    - pyyaml >=6.0
+    - types-pyyaml >=6.0
+    - importlib-metadata  # python <= 3.9
+    - typing_extensions  # python <= 3.10
```
Comment (ocefpaf) [inline recipes/servicex/meta.yaml:39]: Nice to keep the comments! Reminder to remove when we bump the min Python.

**Comment by matthewfeickert** [comment]
Thanks for the help and fast merge @ocefpaf!

**Pull request merged**